packages/babel-parser/src/parser/expression.js

```diff
@@ -801,6 +804,10 @@ export default class ExpressionParser extends LValParser {
         node,
       );
     } else {
+      if (refExpressionErrors !== null) {
+        this.checkExpressionErrors(refExpressionErrors, true);
```

Comment:
nit: Can we move this to the `state.maybeAsyncArrow` branch?

If Flow is complaining, you can add an invariant comment to tell flow that `state.maybeAsyncArrow` is equivalent to `refExpressionErrors !== null`:

```js
/*:: invariant(refExpressionErrors !== null) */
```